framework/source/class/qx/lang/normalize/Array.js

```diff
@@ -251,6 +251,58 @@ qx.Bootstrap.define("qx.lang.normalize.Array", {
 
       return true;
     },
+    
+    /**
+     * The <code>find()</code> method returns a value in the array, if an element in the 
+     * array satisfies the provided testing function. Otherwise undefined is returned.
+     *
+     * <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find">MDN documentation</a> |
+     *
+     * @param callback {Function} Function to test for each element.
+     * @param obj {Object?} Value to use as <code>this</code> when executing <code>callback</code>.
+     * @return {Object} result, undefined if not found
+     */
+    find : function(callback, obj) {
+      if (qx.core.Environment.get("qx.debug")) {
+        qx.core.Assert.assertFunction(callback);
+      }
+      
+      var l = this.length;
+      for (var i = 0; i < l; i++) {
+        var value = this[i];
+        if (callback.call(obj || window, value, i, this)) {
```

Comment:
Let's be careful with the use of *window* since it exists only the browser environment. qooxdoo is also very useful for its OO features, in a backend app, where *window* does not exist. *global* is used as a replacement in nodejs. We may want to make this *obj || window || global*, but should check whether *window* is mapped in the start-up code when using the standalone qooxdoo variant.